core/proc/element_reference_spec.rb

```diff
@@ -14,3 +15,12 @@
 describe "Proc#call on a Proc created with Kernel#lambda or Kernel#proc" do
   it_behaves_like :proc_call_on_proc_or_lambda, :call
 end
+
+describe "Proc#[] with frozen_string_literals" do
```

Comment:
Since it is not backported yet, could you add a ruby_bug guard as shown in CONTRIBUTING.md with range ''...'2.6' ?